src/py/flwr/client/grpc_rere_client/connection.py

```diff
@@ -165,25 +160,14 @@ def send(message: Message) -> None:
         if node_store[KEY_NODE] is None:
             log(ERROR, "Node instance missing")
             return
-        node: Node = cast(Node, node_store[KEY_NODE])
 
         # Get incoming TaskIns
         if state[KEY_TASK_INS] is None:
             log(ERROR, "No current TaskIns")
             return
-        task_ins: TaskIns = cast(TaskIns, state[KEY_TASK_INS])
-
         # Construct TaskRes
         task_res = message_to_taskres(message)
 
-        # Check if fields to be set are not initialized
-        if not validate_task_res(task_res):
-            state[KEY_TASK_INS] = None
-            log(ERROR, "TaskRes has been initialized accidentally")
-
-        # Configure TaskRes
-        task_res = configure_task_res(task_res, task_ins, node)
```

Comment:
It seems like we don't need `state` at all any more, right? We can remove it in this PR, or in a separate one.

Reply:
I think we may want to keep it. We need this in `delete_node`. Plus, I think we need to add a san-check in `receive()` to verify if the consumer node id in received messages is this node.